pkg/engine/watchermanager.go

```diff
@@ -75,7 +75,20 @@ func (r *watcherManager) WatchPackageRevisions(ctx context.Context, filter repos
 	inserted := false
 	for i, watcher := range r.watchers {
 		if watcher != nil {
-			active += 1
+			//Remove any watchers if they are finished and insert the new watcher in the empty slot of the first removed watcher
+			if watcher.isDoneFunction() != nil {
```

Comment:
Below `watcher.isDoneFunction` is returning an `error`. Below it's printed to as an error log. Would it make sense to print it?